### ISSUE TITLE
replace xdg-open & xdg-email

### ISSUE THREAD

**Issue by jscurtu** [jangernert/FeedReader#389]
Now that the missing portal is implemented.. We need to figure out how to implement org.freedesktop.portal.Email.ComposeEmail and org.freedesktop.portal.OpenURI and replace xdg-open and xdg-email in the plugins as they dont work with flatpak and never will.
One possible way would be to make D-Bus calls.

reference
https://github.com/flatpak/xdg-desktop-portal/pull/88
https://github.com/matthiasclasen/portal-test
https://github.com/flatpak/xdg-desktop-portal

**Comment by jangernert** [comment]
I don't believe xdg-email is ever used in the code. What I do is compose a mailto-string and hand it over to xdg-open.
Okay so here are all the use-cases:
- Browser-plugin: just open URL with default browser. This can easily be replaced by show_uri
- Email-plugin: send email
- Utils: write text file to /tmp (is that allowed in flatpak?) and then open the file with default editor
- MediaPopover: open URL to some audio/video in the default media player that can stream it like VLC

**Comment by jangernert** [comment]
Browser-plugin is taken care of ec2d487ae99054b680906c027182c39c35514e9e
Is the email-portal already shipped in a release? At least on my system (Fedora 25) I can't find a dbus-service with that name.

**Comment by jscurtu** [comment]
The email portal is pretty new, I hope it will be shipped very soon in Fedora (which I use as well). 
Unfortunately its not in rawhide or branched on github as a release yet.. 

If it helps us move forward, I can build us an rpm with the latest xdg-desktop-portal API for development. 

I was thinking, we could implement the browser plugin the portal way, then adapt it later to the composeemail portal.. ?

**Comment by jangernert** [comment]
Can we assume that the portal dbus-service is present even for systems that do not support (the latest) flatpak? Otherwise this would potentially break other packages or force them to depend on xdg-desktop-portal.

**Comment by jscurtu** [comment]
the openuri portal dbus-service has been present for a long time, actually that's what gets called when you use Gtk.show_uri_on_window() or Use Gtk.show_uri() when flatpak'd.  

dont get me wrong... I dont want to make xdg-desktop-portal a dependency outside of it other than when using flatpak. 

Just wanted to start a discussion on this, will try this and that and see what I come up with in a own local branch after I get the autobuild on commit up and running ..

**Comment by jscurtu** [comment]
i think we can close this for now..

**Comment by jangernert** [comment]
Btw: apparently shwo_uri should also handle mailto-strings. So no need to spawn xdg-open from command line.

> This is a convenience function for launching the default application to show the uri. The uri must be of a form understood by GIO (i.e. you need to install gvfs to get support for uri schemes such as http:// or ftp://, as only local files are handled by GIO itself). Typical examples are - `file:///home/gnome/pict.jpg` - `http://www.gnome.org` - `mailto:me\@gnome.org`

https://people.gnome.org/~gcampagna/docs/Gtk-3.0/Gtk.show_uri.html

I guess this will be hooked up to the mail-portal once it is finished.

**Comment by jscurtu** [comment]
Awsome find.. thats great, then we just sit and wait. 
Guess we would also need to give access to gvfs.

**Comment by jscurtu** [comment]
https://github.com/jangernert/FeedReader/commit/8613479707e4222287192015528b785572922a4d
